pkg/webhooks/validation/controller/controller.go

```diff
@@ -279,37 +282,76 @@ func (c *Controller) processNextWorkItem() (cont bool) {
 	}
 	defer c.queue.Done(obj)
 
-	req, ok := obj.(*reconcileRequest)
+	req, ok := obj.(reconcileRequest)
 	if !ok {
 		// don't retry an invalid reconcileRequest item
 		c.queue.Forget(req)
 		return true
 	}
 
+	// empty webhook name means we must patch for each webhook
+	if req.webhookName == "" {
+		err := c.updateAll()
+		if err != nil {
+			c.queue.AddRateLimited(reconcileRequest{
+				event:       retryEvent,
+				description: "retry reconcile request",
+				webhookName: "",
+			})
+		}
+		return true
+	}
+
 	if retry, err := c.reconcileRequest(req); retry || err != nil {
-		c.queue.AddRateLimited(retryRequest)
+		c.queue.AddRateLimited(reconcileRequest{
+			event:       retryEvent,
+			description: "retry reconcile request",
+			webhookName: req.webhookName,
+		})
 		utilruntime.HandleError(err)
 	} else {
 		c.queue.Forget(obj)
 	}
 	return true
 }
 
-var retryRequest = &reconcileRequest{
-	event:       retryEvent,
-	description: "retry reconcile request",
+// updateAll updates all webhooks matching the controller's revision, generally in response
+// to a CA bundle update. updateAll reports an error only when there's an issue listing webhooks,
+// not when it has an issue updating a single webhook so that we can retry separately for the
+// two cases.
+func (c *Controller) updateAll() error {
+	whs, err := c.client.AdmissionregistrationV1().ValidatingWebhookConfigurations().List(context.Background(), metav1.ListOptions{})
```

Comment:
I would prefer `c.webhookInformer.GetStore().List()`. Its cached already, so no call to api-server like here, and already has filter logic for `-l istio.io/rev=<rev>`

Reply:
Or if not, we should probably at least filter?

Reply:
Makes sense, fixed